akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -272,12 +274,6 @@ private[stream] object TcpConnectionStage {
 
       override def onUpstreamFailure(ex: Throwable): Unit = {
         if (connection != null) {
-          if (interpreter.log.isDebugEnabled) {
```

Comment:
Note: we discussed this at length via chat 😉 

So now if someone uses the built in idle-timeout in TCP they'll get:

```
[DEBUG] [12/13/2016 18:18:51.971] [ServerTest-akka.actor.default-dispatcher-4] [akka://ServerTest/user/StreamSupervisor-0/flow-1-0-unknown-operation] Aborting tcp connection because of upstream failure: TCP idle-timeout encountered  on connection to [/127.0.0.1:63483], no bytes passed in the last 10 seconds
akka.stream.impl.io.TcpIdleTimeout$$anonfun$2.applyOrElse(TcpStages.scala:377)
akka.stream.impl.io.TcpIdleTimeout$$anonfun$2.applyOrElse(TcpStages.scala:377)
scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
akka.stream.impl.fusing.MapError$$anon$12.onUpstreamFailure(Ops.scala:285)
akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:733)
akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:616)
akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:471)
akka.stream.impl.fusing.GraphInterpreterShell.receive(ActorGraphInterpreter.scala:423)
akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:603)
akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:618)
akka.actor.Actor$class.aroundReceive(Actor.scala:496)
akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:529)
akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
akka.actor.ActorCell.invoke(ActorCell.scala:495)
akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
akka.dispatch.Mailbox.run(Mailbox.scala:224)
akka.dispatch.Mailbox.exec(Mailbox.scala:234)
scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

Which is what @jrudolph argued for, a better message by itself already. I kept the stacktrace as you wanted it.
I'll proceed to changing Akka HTTP to handle the timeouts by itself.